app/models/appeal_submission_upload.rb

```diff
@@ -9,4 +9,16 @@ class AppealSubmissionUpload < ApplicationRecord
           foreign_key: 'guid',
           class_name: 'DecisionReviewEvidenceAttachment',
           inverse_of: :appeal_submission_upload, dependent: :nullify
+
+  scope :failure_not_sent, -> { where(failure_notification_sent_at: nil).order(id: :asc) }
+
+  # Matches all characters except first 3 characters, last 6 characters (2 + .pdf extension), underscores, and hyphens
+  MASK_REGEX = /(?<=.{3})[^_-](?=.{6})/
+
+  def masked_attachment_filename
+    filename = JSON.parse(decision_review_evidence_attachment&.file_data || '{}')['filename']
+    raise 'Filename for not found' if filename.nil?
+
+    filename.gsub(MASK_REGEX, '*')
```

Comment:
I think this should be `X` and not `*` as the VA Notify template uses Markdown and `*` is a special character in Markdown